doc/no-runtime.md

```diff
@@ -15,9 +15,23 @@ If for some reason you want to use your own network layer and don't want to use
     okhttp3.Response httpResponse = ...;
 
     // if you have custom scalar types, provide proper instance of ScalarTypeAdapters with your own custom adapters
-    ScalarTypeAdapters scalarTypeAdapters = new ScalarTypeAdapters(Collections.<ScalarType, CustomTypeAdapter>emptyMap())
+    ScalarTypeAdapters scalarTypeAdapters = new ScalarTypeAdapters(Collections.<ScalarType, CustomTypeAdapter>emptyMap());
 
     Response<Query.Data> response = new Query().parse(httpResponse.body().source(), scalarTypeAdapters);
 ```
 
-Just make sure you added `apollo-api` dependency to your project's build.gradle file.
+To compose a GraphQL POST request along with operation variables to be sent to the server, you can use `Operation.Variables#marshal()` API: 
+
+```java
+    // Generated GraphQL query, mutation, subscription
+    final Query query = ...;
+
+    // if you have custom scalar types, provide proper instance of ScalarTypeAdapters with your own custom adapters
+    final ScalarTypeAdapters scalarTypeAdapters = new ScalarTypeAdapters(Collections.<ScalarType, CustomTypeAdapter>emptyMap());
+
+    final String requestPayload = "{" +
+        "\"operationName\": " + query.name().name() + ", " +
+        "\"query\": " + query.queryDocument() + ", " +
+        "\"variables\": " + query.variables().marshal(scalarTypeAdapters) +
+        "}";
```

Comment:
Should we add a helper function that does this ? Something like:

```
fun Operation.payload(): String
```

?

Reply:
The initial reason why I introduced marshalling for variables only, is to give dev flexibility on deciding how to send GraphQL request to the server: use POST, GET, send just operatation ID, or bundle with bunch of extension attributes etc.

But I see values of having simple case, as most of the time it will be plain http POST request. 

How about if we introduce new API to the `com.apollographql.apollo.api.Operation` like `payload():String` or `postPayload():String` or  `requestPayload()`. Not sure about how to name it, any thoughts?

Reply:
I was suggesting `Operation.payload(): String` but maybe we want to keep `Operation` transport agnostic ? 

In that case, I'm not sure where to put it... Maybe we can just wait to see how users are going to use the API before we add some API that is never going to be used in real life 😅

Reply:
I see. If there are 3 different usages, I would create utility functions (extensions in Kotlin) decoupled from the actual implementation to provide such extra nice to have functionality. 

This could be separate class (maybe in separate package, maybe not). It could even be separate artifact

Reply:
Ok lets leave this as is for now, and we can improve later when we migrate everything to Kotlin.